sdk/core/core-amqp/test/errors.spec.ts

```diff
@@ -20,22 +20,33 @@ class AMQPError {
 
 describe("Errors", function() {
   describe("translate", function() {
-    it("Converts to MessagingError, and acts as a passthrough if the input is not an AmqpProtocolError", function() {
-      const MyError: any = function() {};
-      const err: any = new MyError();
-      const msg: any = undefined;
-      const ehError = new Errors.MessagingError(msg);
-      const translatedError = <Errors.MessagingError>Errors.translate(err);
-      translatedError.name.should.equal(ehError.name);
-      translatedError.retryable.should.equal(ehError.retryable);
-      translatedError.message.should.equal(ehError.message);
+    it("Does not touch errors that are neither AmqpError nor SystemError", function() {
+      const testError = new Error("Test error");
+      const translatedError = Errors.translate(testError);
+      translatedError.name.should.equal(testError.name);
+      translatedError.message.should.equal(testError.message);
+    });
+
+    it("Does not touch TypeError", function() {
+      const testError = new TypeError("This is a wrong type!!");
+      const translatedError = Errors.translate(testError);
+      translatedError.name.should.equal(testError.name);
+      translatedError.message.should.equal(testError.message);
+    });
+
+    it("Does not touch RangeError", function() {
+      const testError = new RangeError("Out of range!!");
+      const translatedError = Errors.translate(testError);
+      translatedError.name.should.equal(testError.name);
+      translatedError.message.should.equal(testError.message);
```

Comment:
Since the above 3 error cases should go untouched, can we do a deep check to ensure that the input and the output for `translate()` is exactly the same?

Reply:
Updated to use deep equality.